fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java

```diff
@@ -476,8 +479,16 @@ protected OptExpression createUnion(OptExpression queryInput, OptExpression view
             logMVRewrite(mvRewriteContext, "Rewrite aggregate with union failed: rewrite aggregate for union failed");
             return null;
         }
-        return createNewAggregate(rewriteContext, queryAgg, newAggregations, aggregateMapping,
+        OptExpression result = createNewAggregate(rewriteContext, queryAgg, newAggregations, aggregateMapping,
                 unionExpr, newProjection);
+        if (rewriteContext.getUnionRewriteQueryExtraPredicate() != null) {
```

Comment:
That's pretty hack

Reply:
In this case, I have no better idea for now. 

Just like the retry policy, when rewrite found union-all failed, retry to pull up some predicates to rewrite again.

Maybe we can introduce a pull-up normalized rule for a better-normalized query rewrite predicates that only match mv's expression to avoid the retry.

But this may affect the rewrite framework, we can do it later.